File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        
        Deque<Integer> progressesQueue = new ArrayDeque<>();
        Deque<Integer> speedsQueue = new ArrayDeque<>();
        for(int p: progresses){
            progressesQueue.add(p);
        }
        for(int s: speeds){
            speedsQueue.add(s);
        }
        // System.out.println(progressesQueue);
        
        while(!progressesQueue.isEmpty()){
                int answerTemp=0;
                int day=0;
                int firstProgress = (int)progressesQueue.toArray()[0];
                int firstSpeed = (int)speedsQueue.toArray()[0];
                
                while(firstProgress<100){
                    firstProgress+=firstSpeed;
                    day++;
                }
                progressesQueue.poll();
                speedsQueue.poll();
                answerTemp++;
                
                while(!progressesQueue.isEmpty()){
                    int anotherProgress = (int)progressesQueue.toArray()[0];
                    int anotherSpeed = (int)speedsQueue.toArray()[0];
                    if(anotherProgress+anotherSpeed*day>=100){
                        progressesQueue.poll();
                        speedsQueue.poll();
                        answerTemp++;
                    }else{
                        break;
                    }
                }
                answer.add(answerTemp);
            
        }
        // System.out.println(answer);
        return answer.stream().mapToInt(i->i).toArray();
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    
    let now = 0;
    
    const process = progresses.map((value, index)=>{
        const temp = 100-value;
        const temp2 = temp / speeds[index]
        return Math.ceil(temp2)
    })
    
    let count = 1;
    for(let i=1; i<process.length; i++){
        if(process[now]>=process[i]){
            count+=1;
        }else{
            answer.push(count)
            count=1;
            now=i;
        }
    }
    
    answer.push(count)
    
    return answer;
}